### PR TITLE
Migration detection and migration fragment added.

### PR DESCRIPTION
**Pull request by monday8am**

### CODE REVIEW

File: Joey/UI/Activities/MainDrawerActivity.cs [hunk 1]
```diff
@@ -186,6 +165,56 @@ public override void OnBackPressed()
...
+        }
+
+        private bool tryMigrateDatabase(IUserData userData)
+        {
+            var oldVersion = DatabaseHelper.CheckOldDb(DatabaseHelper.GetDatabaseDirectory());
+            if (oldVersion == -1)
+                return false;
+
+            var migrationFragment = MigrationFragment.Init(oldVersion, SyncSqliteDataStore.DB_VERSION);
+            OpenFragment(migrationFragment);
```
Comment (monday8am) [inline Joey/UI/Activities/MainDrawerActivity.cs]: @alfonsogarciacaro I have added the migration detection methods and a dummy migration fragment. You can use it if you want to implement the migration screen.

File: Joey/UI/Fragments/FeedbackNoApiFragment.cs [hunk 2]
```diff
@@ -12,6 +12,14 @@ public class FeedbackNoApiFragment : Fragment
...
         private Button noUserRegisterButton;
 
+        public FeedbackNoApiFragment()
+        {
+        }
+
+        public FeedbackNoApiFragment(IntPtr jref, Android.Runtime.JniHandleOwnership xfer) : base(jref, xfer)
+        {
+        }
+
```
Comment (taavirehemagi) [inline Joey/UI/Fragments/FeedbackNoApiFragment.cs]: Are these constructors necessary? they are not present any more in LogTimeEntryListFragment for example...

File: Joey/UI/Adapters/DrawerListAdapter.cs [hunk 3]
```diff
@@ -191,7 +183,7 @@ protected override void Rebind()
         public enum VisibilityMode
         {
             Normal,
-            Offline,
+            NoApiToken,
```
Comment (taavirehemagi) [inline Joey/UI/Adapters/DrawerListAdapter.cs]: In this case the wording "noApiToken" seems to specific and doesn't capture the meaning of the state. Maybe Connected and NotConnected over "Normal" and "Offline". Also, what if in the future we will use a different way of determining if a user is connected or not?

File: Joey/UI/Fragments/FeedbackNoApiFragment.cs (hunk 2 above)
Comment (monday8am) [inline Joey/UI/Fragments/FeedbackNoApiFragment.cs]: just check this commit from 15 days ago:
https://github.com/toggl/mobile/commit/932696f3511baae984d3b18ebb396f6f0e917341

or the same file in master:
https://github.com/toggl/mobile/blob/master/Joey/UI/Fragments/LogTimeEntriesListFragment.cs

Or this search in XI:
https://insights.xamarin.com/app/Toggl-Android/issues?s=Unable%20to%20activate%20instance%20of%20type%20

Comment (taavirehemagi) [inline Joey/UI/Fragments/FeedbackNoApiFragment.cs]: alright. if it's still necessary then good with me 👍  I just didn't see any examples in unidirectional mode anymore and figured that the need for this is over. fine with me then

**Pull request merged**